testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/management/ServerAutoStartTestCase.java

```diff
@@ -0,0 +1,281 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILURE_DESCRIPTION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SERVER_GROUP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.START_SERVERS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.STOP_SERVERS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.test.integration.domain.management.util.DomainTestUtils.waitUntilState;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.client.helpers.domain.DomainClient;
+import org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil;
+import org.jboss.as.test.integration.domain.management.util.DomainTestSupport;
+import org.jboss.dmr.ModelNode;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Emanuel Muckenhuber
+ */
+public class ServerAutoStartTestCase {
+
+    private static final String STOPPED_EXT = ".stopped";
+    private static final String STARTED_EXT = ".started";
+
+    private static DomainTestSupport testSupport;
+    private static DomainLifecycleUtil domainMasterLifecycleUtil;
+    private static DomainLifecycleUtil domainSlaveLifecycleUtil;
+    private static final ModelNode hostMaster = new ModelNode();
+    private static final ModelNode hostSlave = new ModelNode();
+    private static final ModelNode mainOne = new ModelNode();
+    private static final ModelNode mainTwo = new ModelNode();
+    private static final ModelNode mainThree = new ModelNode();
+    private static final ModelNode mainFour = new ModelNode();
+    private static final ModelNode otherOne = new ModelNode();
+    private static final ModelNode otherTwo = new ModelNode();
+    private static final ModelNode otherThree = new ModelNode();
+    private static final ModelNode otherFour = new ModelNode();
+    private static final Path autoStartMasterDataDir
+            = Paths.get(DomainTestSupport.getHostDir(ServerAutoStartTestCase.class.getSimpleName(), "master").toURI())
+            .resolve("data")
+            .resolve("auto-start");
+    private static final Path autoStartSlaveDataDir
+            = Paths.get(DomainTestSupport.getHostDir(ServerAutoStartTestCase.class.getSimpleName(), "slave").toURI())
```

Comment:
`File.toPath()` would be a better approach than creating a new `Path` based on the URI.